### PR TITLE
Code Quality: Improved app startup routine 2

### PR DESCRIPTION
**Pull request by d2dyno1**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #issue...

**Validation**
How did you test these changes?
- [ ] Did you build the app and test your changes?

### CODE REVIEW

**Comment by 0x5bfa** [comment]
Are you planning to convert instancing helpers into services and removing instanced helpers in App.App?

**Comment by yaira2** [comment]
@d2dyno1 what's the plan for this PR?

**Comment by d2dyno1** [comment]
I'm going to continue work on it this Thursday 🙂

**Comment by d2dyno1** [comment]
Ready for review!

**Review by 0x5bfa — APPROVED** [review]
Great! in terms of codebase quality.

I understand as follows

- Update app UI thread initialization stage
- Rename ImageModel to Image
- Add a service for pinning/unpinning
- System.IO to SystemIO (FYI, I introduced this simpified namespace alias to see all reference to that namespace because this namespace reference in storage class is not favorable)

File: src/Files.Core/Services/ISystemPinService.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using Files.Core.Storage;
+
+namespace Files.Core.Services
+{
+	/// <summary>
+	/// A service that allows to manage pinning and unpinning items in common OS places.
```
Comment (yaira2) [inline src/Files.Core/Services/ISystemPinService.cs:6]: Is this specific to the start menu, or does it refer to the sidebar as well?

File: src/Files.Shared/Utils/IImage.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.Shared.Utils
+{
+	/// <summary>
+	/// Represents an image which can be displayed in the UI.
```
Comment (yaira2) [inline src/Files.Shared/Utils/IImage.cs:7]: What type of image, is this for thumbnails/icons?

File: src/Files.Core/Services/ISystemPinService.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	{
+		// TODO(s)
+		[Obsolete("Use IsPinnedAsync instead. This method is used for a workaround in ListedItem class to avoid major refactoring.")]
+		bool IsPinned(string folderPath);
+
+		Task<bool> IsPinnedAsync(IFolder folder);
+
+		Task PinAsync(IFolder folder);
+
+		Task UnpinAsync(IFolder folder);
```
Comment (yaira2) [inline src/Files.Core/Services/ISystemPinService.cs:18]: Can you add docs?

File: src/Files.App/ViewModels/MainPageViewModel.cs [hunk 4]
```diff
@@ -277,16 +276,12 @@ public async Task<(string tabLocationHeader, IconSource tabIcon, string toolTipT
 			return (tabLocationHeader, iconSource, toolTipText);
 		}
 
-		public async Task OnNavigatedTo(NavigationEventArgs e)
+		public async Task OnNavigatedTo(object? parameter)
 		{
-			if (e.NavigationMode == NavigationMode.Back)
-				return;
-
```
Comment (yaira2) [inline src/Files.App/ViewModels/MainPageViewModel.cs:284]: Why was this removed?

File: src/Files.App/Data/Models/AppModel.cs [hunk 5]
```diff
@@ -37,20 +38,12 @@ public int TabStripSelectedIndex
...
-					Frame rootFrame = (Frame)MainWindow.Instance.Content;
-					var mainView = (MainPage)rootFrame.Content;
+					var mainView = (MainPage)MainWindow.Instance.RootControl.AppContent;
 					mainView.ViewModel.SelectedTabItem = MainPageViewModel.AppInstances[value];
 				}
 			}
 		}
 
-		private bool isAppElevated = false;
-		public bool IsAppElevated
```
Comment (yaira2) [inline src/Files.App/Data/Models/AppModel.cs:48]: Is there a replacement?

File: src/Files.Core/Services/ISystemPinService.cs (hunk 1 above)
Comment (d2dyno1) [reply src/Files.Core/Services/ISystemPinService.cs:6]: It is the refactored `SecondaryTileHelper` class. The functionality remains unchanged.

File: src/Files.Shared/Utils/IImage.cs (hunk 2 above)
Comment (d2dyno1) [reply src/Files.Shared/Utils/IImage.cs:7]: Any image. This is a common interface for all images - the implementation holds the necessary information which is all managed by the frontend (UI project). The backend is only exposed to the `IImage` interface and has no internal knowledge of the imaging implementation

File: src/Files.App/Data/Models/AppModel.cs (hunk 5 above)
Comment (d2dyno1) [reply src/Files.App/Data/Models/AppModel.cs:48]: It was not used anywhere

File: src/Files.Core/Services/ISystemPinService.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.Core/Services/ISystemPinService.cs:6]: It's a little generic right now, can we clarify in the name that it's for the start menu?

File: src/Files.Core/Services/ISystemPinService.cs (hunk 3 above)
Comment (d2dyno1) [reply src/Files.Core/Services/ISystemPinService.cs]: Done

File: src/Files.Core/Services/ISystemPinService.cs (hunk 1 above)
Comment (d2dyno1) [reply src/Files.Core/Services/ISystemPinService.cs]: Renamed to `IStartMenuService`

File: src/Files.App/ViewModels/MainPageViewModel.cs (hunk 4 above)
Comment (d2dyno1) [reply src/Files.App/ViewModels/MainPageViewModel.cs:284]: I don't see any cases where the (whole) app view would be navigated backward. We only navigated to the splash screen and then the main view

**Comment by hishitetsu** [comment]
This appears to prevent launch from a cached instance from working. When I try to launch the app again, the window does not appear.

**Comment by yaira2** [comment]
Should we revert?

**Comment by hishitetsu** [comment]
Before we revert, I first want to figure out how to make launching from the cached instance work again.

**Comment by hishitetsu** [comment]
Hmmm, even when I get the window to show up, the tabs don't show up properly or the app crashes.
If it is not required for the V3 release, it would be better to revert it once and test it after the V3 release.
